packages/beacon-node/src/chain/opPools/aggregatedAttestationPool.ts

```diff
@@ -96,16 +121,20 @@ export class AggregatedAttestationPool {
       return InsertOutcome.Old;
     }
 
-    const attestationGroupByDataHashByIndex = this.attestationGroupByDataHashByIndexBySlot.getOrDefault(slot);
-    let attestationGroupByDataHash = attestationGroupByDataHashByIndex.get(attestation.data.index);
-    if (!attestationGroupByDataHash) {
-      attestationGroupByDataHash = new Map<DataRootHex, MatchingDataAttestationGroup>();
-      attestationGroupByDataHashByIndex.set(attestation.data.index, attestationGroupByDataHash);
+    const attestationGroupByIndexByDataHash = this.attestationGroupByIndexByDataHashBySlot.getOrDefault(slot);
+    let attestationGroupByIndex = attestationGroupByIndexByDataHash.get(dataRootHex);
+    if (!attestationGroupByIndex) {
+      attestationGroupByIndex = new Map<CommitteeIndex, MatchingDataAttestationGroup>();
+      attestationGroupByIndexByDataHash.set(dataRootHex, attestationGroupByIndex);
     }
-    let attestationGroup = attestationGroupByDataHash.get(dataRootHex);
+    const committeeIndex = isElectraAttestation(attestation)
+      ? // this attestation is added to pool after validation
+        (attestation.committeeBits.getSingleTrueBit() as number)
+      : attestation.data.index;
+    let attestationGroup = attestationGroupByIndex.get(committeeIndex);
     if (!attestationGroup) {
       attestationGroup = new MatchingDataAttestationGroup(committee, attestation.data);
-      attestationGroupByDataHash.set(dataRootHex, attestationGroup);
+      attestationGroupByIndex.set(committeeIndex, attestationGroup);
```

Comment:
`committeeIndex` can be null if there is more than one bit set in `committeeBits`. I think we should handle the null case to void having null key in `attestationGroupByIndex`

Reply:
all attestations should be validated before reaching this pool. Will throw error in that case